android/src/main/java/com/facebook/flipper/plugins/network/FlipperOkhttpInterceptor.java

```diff
@@ -6,46 +6,89 @@
  */
 package com.facebook.flipper.plugins.network;
 
+import android.text.TextUtils;
+import com.facebook.flipper.core.*;
+import com.facebook.flipper.plugins.common.BufferingFlipperPlugin;
 import com.facebook.flipper.plugins.network.NetworkReporter.RequestInfo;
 import com.facebook.flipper.plugins.network.NetworkReporter.ResponseInfo;
-import java.io.IOException;
-import java.util.ArrayList;
-import java.util.List;
-import java.util.Set;
-import java.util.UUID;
-import javax.annotation.Nullable;
-import okhttp3.Headers;
-import okhttp3.Interceptor;
-import okhttp3.Request;
-import okhttp3.Response;
-import okhttp3.ResponseBody;
+import okhttp3.*;
 import okio.Buffer;
 import okio.BufferedSource;
 
-public class FlipperOkhttpInterceptor implements Interceptor {
+import javax.annotation.Nullable;
+import java.io.IOException;
+import java.util.*;
+
+public class FlipperOkhttpInterceptor implements Interceptor, BufferingFlipperPlugin.ConnectionListener {
 
   public @Nullable NetworkFlipperPlugin plugin;
 
+  private Map<String, ResponseInfo> mockResponeMap = new HashMap<>();
+
   public FlipperOkhttpInterceptor() {
     this.plugin = null;
   }
 
   public FlipperOkhttpInterceptor(NetworkFlipperPlugin plugin) {
     this.plugin = plugin;
+    this.plugin.setConnectionListener(this);
+
+    // TODO test code, remove it later
+//    ResponseInfo mockResponse = new ResponseInfo();
+//    mockResponse.body = "TEST".getBytes();
+//    mockResponse.statusCode = 200;
+//    mockResponse.statusReason = "OK";
+//    registerMockResponse("https://api.github.com/repos/facebook/yoga", "GET", mockResponse);
+  }
+
+  protected void registerMockResponse(String requestUrl, String method, ResponseInfo response) {
+    mockResponeMap.put(requestUrl + "|" + method, response);
   }
 
   @Override
   public Response intercept(Interceptor.Chain chain) throws IOException {
     Request request = chain.request();
     String identifier = UUID.randomUUID().toString();
     plugin.reportRequest(convertRequest(request, identifier));
-    Response response = chain.proceed(request);
+
+    // Check if there is a mock response
+    Response mockResponse = getMockResponse(request);
+    Response response = mockResponse != null ? getMockResponse(request) : chain.proceed(request);
```

Comment:
nit: `Response response = mockResponse != null ? mockResponse : chain.proceed(request);`

Reply:
Sure thing, I will update the PR in a bit